File: excelium-model/src/main/java/excelium/model/enums/WriteMode.java
/*
 * MIT License
 *
 * Copyright (c) 2018 Excelium
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package excelium.model.enums;

/**
 * Represents mode of test data writing.
 *
 * @author PhungDucKien
 * @since 2018.05.25
 */
public enum WriteMode {
    /**
     * Append write mode.
     * Inserts and updates records without effecting others.
     */
    APPEND,
    /**
     * Replace write mode.
     * Truncates the table before inserting records.
     */
    REPLACE,
    /**
     * Remove write mode.
     * Removes records without effecting others.
     */
    REMOVE,
    /**
     * Truncate write mode.
     * Truncates the table.
     */
    TRUNCATE;

    /**
     * From name write mode.
     *
     * @param name the name
     * @return the write mode
     */
    public static WriteMode fromName(String name) {
        for (WriteMode w : WriteMode.values()) {
            if (w.name().equalsIgnoreCase(name)) {
                return w;
            }
        }
        return APPEND;
    }
}

File: packages/model/src/enums/WriteMode.ts
/*
 * Licensed to the Software Freedom Conservancy (SFC) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The SFC licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

/**
 * Represents mode of test data writing.
 *
 * @author PhungDucKien
 * @since 2018.05.25
 */
enum WriteMode {
  /**
   * Append write mode.
   * Inserts and updates records without effecting others.
   */
  APPEND,
  /**
   * Replace write mode.
   * Truncates the table before inserting records.
   */
  REPLACE,
  /**
   * Remove write mode.
   * Removes records without effecting others.
   */
  REMOVE,
  /**
   * Truncate write mode.
   * Truncates the table.
   */
  TRUNCATE,
}

namespace WriteMode {
  /**
   * From name write mode.
   *
   * @param name the name
   * @return the write mode
   */
  export function fromName(name: string): WriteMode {
    let writeMode = (WriteMode as any)[name.toUpperCase()];
    if (!writeMode) {
      writeMode = WriteMode.APPEND;
    }
    return writeMode;
  }
}

export default WriteMode;
